### PR TITLE
Feature/onboarding

### PR DESCRIPTION
**Pull request by emilijadunoska**
### Description
This PR is for the development of the Onboarding System for ESID that effectively guides new users through the application's features and helps them become familiar with it. 

#### Related Issues
- #370 
- #372
- #373

### Design Decisions
The onboarding system is developed with the React joyride library. There is also a welcome modal that is displayed during the user's initial visit which provides an overview of the application through a series of informative slides and illustrations. The onboarding tours are accessible through the information button in the top bar, which makes them available without being intrusive. Users have the flexibility to start, pause, or retake any tour at their own pace. This implementation makes the onboarding experience adaptable for both new and existing users. 

Possible alternative implementations were considered, such as creating a one-time static tour that users would go through only during their first visit. However, this approach is less flexible and doesn't allow users to revisit the tours later or learn at their own pace. 

### Checklist

**I, the author of this PR checked the following requirements for good software quality:**

- [x] The code is properly formatted (I ran the formatter)
- [x] The code is written with our software quality standards (I ran the linter)
- [x] The code is written using our code style
- [x] Extensive in source documentation has been added
- [x] Unit and/or integration tests have been added
- [x] All texts have been internationalized with at least the following languages:
  - [x] English
  - [x] German
- [x] I tried addressing all new accessibility problems displayed in the console and documented if they can't be fixed
- [x] I attached performance measurements to prevent performance degradation
- [x] I added the changes to the next release section of the [changelog](../frontend/docs/changelog/changelog-en.md)

**I, the reviewer checked the following things:**

- [ ] I ran the software once and tried all new and related functionality to this PR
- [ ] I looked at all new and changed lines of code and commented on possible problems
- [ ] I read the added documentation and checked if it is understandable and clear
- [ ] I checked the added tests for completeness
- [ ] I checked the internationalized strings for spelling errors
- [ ] I checked the performance metrics for problems or unexplained degradation
- [ ] I checked that the changes are noted in the [changelog](../frontend/docs/changelog/changelog-en.md)

### CODE REVIEW

File: frontend/.env.template
Comment (JonasGilg) [inline frontend/.env.template:1]: The code here shoudn't be removed.

File: frontend/docs/changelog/changelog-de.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ SPDX-License-Identifier: CC-BY-4.0
 ### Neue Funktionen
 
 - Die Daten von Köln werden nun mit einer Auflösung in Stadtteilen angezeigt
+- Onboarding-System für ESID. Es besteht aus:
+  - Willkommensmodal, das dem Benutzer beim ersten Besuch einen Überblick über die ESID anhand von Folien und Illustrationen zeigt
```
Comment (JonasGilg) [inline frontend/docs/changelog/changelog-de.md:16]:
```suggestion
  - Willkommensmodal, das dem Benutzer beim ersten Besuch einen Überblick über ESID anhand von Folien und Illustrationen zeigt
```

File: frontend/src/components/OnboardingComponents/TourComponents/TourSteps.tsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+/**
+ * Interface for the state of the tour steps
+ */
+interface State {
+  /**a flag to run or stop the tour */
+  run: boolean;
+  /**array of Step objects for the tour */
+  steps: Step[];
+  /**the current step index of the tour, keeping state of this value is necessary for joyride controlled (interactive) tours */
```
Comment (JonasGilg) [inline frontend/src/components/OnboardingComponents/TourComponents/TourSteps.tsx:21]:
```suggestion
  /** a flag to run or stop the tour */
  run: boolean;
  /** array of Step objects for the tour */
  steps: Step[];
  /** the current step index of the tour, keeping state of this value is necessary for joyride controlled (interactive) tours */
```

File: frontend/src/components/OnboardingComponents/WelcomeDialog/WelcomeDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            nextButton={
+              step < numberOfSlides - 1 ? (
+                <Button onClick={onNext} data-testid='arrow-forward-button'>
+                  <ArrowForwardIos />
+                </Button>
+              ) : (
+                <Button disabled data-testid='arrow-forward-button'>
+                  <ArrowForwardIos />
+                </Button>
+              )
```
Comment (JonasGilg) [inline frontend/src/components/OnboardingComponents/WelcomeDialog/WelcomeDialog.tsx:143]: Can you move this to a function/callback?

File: frontend/src/components/OnboardingComponents/WelcomeDialog/WelcomeDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            backButton={
+              step > 0 ? (
+                <Button onClick={onPrev} data-testid='arrow-backward-button'>
+                  <ArrowBackIos />
+                </Button>
+              ) : (
+                <Button disabled data-testid='arrow-backward-button'>
+                  <ArrowBackIos />
+                </Button>
+              )
```
Comment (JonasGilg) [inline frontend/src/components/OnboardingComponents/WelcomeDialog/WelcomeDialog.tsx:154]: Can you move this to a function/callback?

**Review by JonasGilg — CHANGES_REQUESTED** [review]
Looks very good. The code quality is very high.

Apart from the comments and suggestions I already made the only other change request would be to suggest, for you to write all comments with a capital letter at the start and if they form a sentence also with a period at the end.

**Comment by emilijadunoska** [comment]
> Looks very good. The code quality is very high.
> 
> Apart from the comments and suggestions I already made the only other change request would be to suggest, for you to write all comments with a capital letter at the start and if they form a sentence also with a period at the end.

Thank you for the positive feedback! I've addressed the comments and suggestions you made.

**Review by JonasGilg — APPROVED** [review]

**Pull request merged**